<?xml version="1.0"?>
<launch>
  <env name="ROSCONSOLE_FORMAT" value="[${node}][${time:%H:%M:%S%F}][${severity}]:${message}"/>

  <group ns="swarm158">
    <node pkg="dithas" type="plane_segmentation" name="dithas_detector" output="log">
      <param name="ransac_dis_thres" value="0.1"/>
      <remap from="~point_cloud" to="/cloud_registered"/>
      <remap from="~plane_pose" to="/swarm158/dithas/tag_pose"/>
    </node>
  </group>

  <group ns="swarm155">
    <node pkg="dithas" type="dithas_path" name="dithas_detector" output="screen">
      <rosparam file="$(find dithas)/config/body2cam.yaml" command="load"/>
      <remap from="~tag_detections" to="/tag_detections"/>
      <remap from="~tag_pose" to="/swarm155/dithas/tag_pose"/>
    </node>
    <node pkg="dithas" type="dithas_pose_solve" name="dithas_solver" output="screen">
      <remap from="~tag_pose_self" to="/swarm155/dithas/tag_pose"/>
      <remap from="~tag_pose_glbl" to="/swarm158/dithas/tag_pose"/>
    </node>
  </group>


  <node pkg="tf" type="static_transform_publisher" name="tf_plane" args="0 0 0 0 0 0 1 /camera_init /map 100" output="screen"/>

  <node pkg="rviz" type="rviz" name="vis_bags" args="-d $(find dithas)/config/lidar_cam.rviz" output="screen"/>

</launch>
